### PR TITLE
Update concept.md

### PR DESCRIPTION
**Pull request by nmiculinic**
As discussed during the bulward sync meetings the following changes to concept document are introduced:

* Organization names unique per bulward organization, project names unique per organization

It's not an issue with any stakeholder and simplifies the implementation. 

Cascading permission deletion issue after the discussion is decided to be more trouble than it's worth. For example, let's say user is given permission to do X. He sets up CI/CD pipeline with a dedicated service account doing X, with its own role and rolebinding. Sometime later user's permission doing X is revoked; here the cascading effect could have unforeseen consequences in breaking the CI/CD pipeline, one which original deletion hasn't thought of. Other major service providers, like AWS/GCP aren't doing cascading permission revocation, thus we removed this from TBD issues. 

```release-note
NONE
```

### CODE REVIEW

**Comment by nmiculinic** [comment]
> Instead of removing them directly from concept doc, how about we move them to a session called Decisions or something like that?

This is alpha/small/greenfield project, and I'd simplify the usual process for more lightweight one, especially for decisions which weren't previously heavily discussed or vetted (like the one I'm amending, where one person wrote it, and it got mostly ignored until it presented an issue)

**Comment by thetechnick** [comment]
So we have two main discussion topics here:

- unique vs. non-unique organization names
- RBAC "cleanup"

## RBAC Cleanup
RBAC Cleanup is discussed in detail here: https://github.com/kubermatic/bulward/pull/66

I am considering proper RBAC cleanup as a core feature of Bulward, without that an integration with KubeCarrier is not possible.

## unique vs. non-unique organization names

Only downside of unique organization names is that users cannot know about collisions, before trying to create a new organization.

Otherwise unique names make stuff easier. I guess the biggest issue would have been that users can be part of multiple Organizations with the same name.
And users can still set a non-unique name via the displayName property.

## Changing concepts

I am not 100% sure, how we want to handle concept changes, but I think retroactively removing parts of an initial concept (or editing it in any substantial manner) can lead to us loosing background/context information.

For me the concepts are incremental design documents, describing intend and it's context.
While the current state of the project (e.g. how it was really implemented) is part of the documentation.

Thats why concepts are usually not up-to-date with all decisions that are made along the way...

tldr; I would like to leave merged design documents alone and submit new ones for bigger changes/features. (In this case the most specific concept document wins out)

**Comment by nmiculinic** [comment]
> RBAC Cleanup

this was created before we had that proposal. 

> unique vs. non-unique organization names

we made a decision for unique organization names on a meeting. This is about documenting such decision even if we rollback them later. 

> Changing concepts

then the concepts have to be date pinned and versioned (ok, they are kinda date pinned and versioned with git). But it's already out of date and we're missing a updated source of truth what the concept is. Also, notice the concept document is in the `docs` folder. 


Whatever we decide I'm hoping we can remove this item from WIP, either closing it or merging it in some form.
